autometa/common/external/diamond.py

```diff
@@ -28,204 +28,359 @@
+    tmpdir : str, optional
+        Path to temporary directory
+        '</path/to/temporary/directory>' by default tempfile.mkdtemp()
+    force : bool, optional
+        overwrite existing diamond results, by default False
+    verbose : bool, optional
+        log progress to terminal, by default False
 
     Returns
     -------
     str
-        `outfpath`
+        Path to BLAST results
 
     Raises
-    -------
+    ------
     FileNotFoundError
         `fasta` file does not exist
     ValueError
         provided `blast_type` is not 'blastp' or 'blastx'
-    OSError
-        Diamond execution failed
+    subprocess.CalledProcessError
+        Failed to run blast
     """
     if not os.path.exists(fasta):
         raise FileNotFoundError(fasta)
     if os.path.exists(outfpath) and not force:
-        empty = not os.stat(outfpath).st_size
-        if not empty:
+        if not os.path.getsize(outfpath):
             if verbose:
-                logger.warning(f'FileExistsError: {outfpath}. To overwrite use --force')
+                logger.warning(f"FileExistsError: {outfpath}. To overwrite use --force")
             return outfpath
     blast_type = blast_type.lower()
-    if blast_type not in ['blastp','blastx']:
-        raise ValueError(f'blast_type must be blastp or blastx. Given: {blast_type}')
+    if blast_type not in ["blastp", "blastx"]:
+        raise ValueError(f"blast_type must be blastp or blastx. Given: {blast_type}")
     if verbose:
-        logger.debug(f'Diamond{blast_type.title()} {fasta} against {database}')
+        logger.debug(f"Diamond{blast_type} {fasta} against {database}")
     cmd = [
-        'diamond',
-        'blastp',
-        '--query',
+        "diamond",
+        blast_type,
+        "--query",
         fasta,
-        '--db',
+        "--db",
         database,
-        '--evalue',
+        "--evalue",
         evalue,
-        '--max-target-seqs',
+        "--max-target-seqs",
         maxtargetseqs,
-        '--threads',
+        "--threads",
         cpus,
-        '--outfmt',
-        '6',
-        '--out',
+        "--outfmt",
+        "6",
+        "--out",
         outfpath,
-        '--tmpdir',
-        tmpdir]
+        "--tmpdir",
+        tmpdir,
+    ]
+    # this is done as when cmd is a list each element should be a string
     cmd = [str(c) for c in cmd]
     if verbose:
         logger.debug(f'RunningDiamond: {" ".join(cmd)}')
-    with open(os.devnull, 'w') as stdout, open(os.devnull, 'w') as stderr:
-        proc = subprocess.run(cmd, stdout=stdout, stderr=stderr)
-    if proc.returncode:
-        raise OSError(f'DiamondFailed:\nArgs:{proc.args}\nReturnCode:{proc.returncode}')
+    proc = subprocess.run(
+        cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=False
+    )
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise err
```

Comment:
1. Similar comments as in the `makedatabase` function.
2. `shell=False` is the default and not necessary

```suggestion
    subprocess.run(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True)
```